src/header.rs

```diff
@@ -1614,15 +1614,34 @@ pub fn bytes2path(bytes: Cow<[u8]>) -> io::Result<Cow<Path>> {
 pub fn bytes2path(bytes: Cow<[u8]>) -> io::Result<Cow<Path>> {
     Ok(match bytes {
         Cow::Borrowed(bytes) => {
-            Cow::Borrowed({ Path::new(str::from_utf8(bytes).map_err(invalid_utf8)?) })
-        }
-        Cow::Owned(bytes) => {
-            Cow::Owned({ PathBuf::from(String::from_utf8(bytes).map_err(invalid_utf8)?) })
+            Cow::Borrowed(Path::new(str::from_utf8(bytes).map_err(invalid_utf8)?))
         }
+        Cow::Owned(bytes) => Cow::Owned(PathBuf::from(
+            String::from_utf8(bytes).map_err(invalid_utf8)?,
+        )),
     })
 }
 
 #[cfg(target_arch = "wasm32")]
 fn invalid_utf8<T>(_: T) -> io::Error {
     io::Error::new(io::ErrorKind::InvalidData, "Invalid utf-8")
 }
+
+pub fn canonicalize(path: &Path) -> PathBuf {
+    canonicalize_or_err(path).unwrap_or(path.to_path_buf())
+}
+
+#[cfg(not(target_os = "wasi"))]
```

Comment:
I don't think that `#[cfg]` is necessary here, I think it's fine to use `if cfg!(...)` and can you also add a comment for why this is specialized on wasi?

Reply:
Refactored the code to use if cfg!(...) instead as you suggested